src/dfx/src/commands/remote/generate_binding.rs

```diff
@@ -56,25 +53,25 @@ pub fn exec(env: &dyn Environment, opts: GenerateBindingOpts) -> DfxResult {
                     );
                     continue;
                 }
-                if main_path.exists() {
+                if main.exists() {
                     if opts.overwrite {
                         info!(
                             log,
                             "Overwriting main file {} of canister {}.",
-                            main,
+                            main.display(),
```

Comment:
I think we default to `to_string_lossy` in this codebase (for no major reason I can remember).